lib/eosknowledgeprivate/ekn-media-bin.c

```diff
@@ -622,10 +623,37 @@ ekn_media_bin_init_video_sink (EknMediaBin *self)
   if (video_sink)
     {
       g_object_set (priv->play, "video-sink", video_sink, NULL);
-      priv->video_sink = g_object_ref (video_sink);
+      priv->video_sink = gst_object_ref_sink (video_sink);
     }
 }
 
+static inline void
+ekn_media_bin_deinit_video_sink (EknMediaBin *self)
+{
+  EknMediaBinPrivate *priv = EMB_PRIVATE (self);
+
+  /* Stop Playback to give gst a chance to cleanup its mess */
+  if (priv->play)
+    gst_element_set_state (priv->play, GST_STATE_NULL);
+
+  /* Stop bus watch */
+  if (priv->bus)
+    {
+      gst_bus_set_flushing (priv->bus, TRUE);
+      gst_bus_remove_watch (priv->bus);
+      g_clear_pointer (&priv->bus, g_object_unref);
```

Comment:
I meant to use gst_object_unref() instead of regular g_object_unref()
In any case I will replace it with gst_object_replace()